### PR TITLE
[Carbon Calculation] One Click LCA UI Widget

### PR DESCRIPTION
**Pull request by bingjiez**
Viewer widget to trigger one-click-LCA carbon calculation jobs. The API behind is provided by reporting platform and calculations are based on model insight reports.

UI mainly includes a table of reports, job trigger button, one-click-LCA login UI, job status report and result (link to report/fail message).

### CODE REVIEW

File: packages/itwin/one-click-lca-widget/README.md
Comment (calebmshafer) [inline packages/itwin/one-click-lca-widget/README.md:4]: Can you add a space prior to this line?

File: packages/itwin/one-click-lca-widget/package.json
Comment (calebmshafer) [inline packages/itwin/one-click-lca-widget/package.json:13]: Can you update the link to `/imodeljs/viewer-components-react` -> `/itwin/viewer-components-react/`

File: packages/itwin/one-click-lca-widget/src/api/oneClickLCA/generated/index.ts
Comment (calebmshafer) [inline packages/itwin/one-click-lca-widget/src/api/oneClickLCA/generated/index.ts:5]: Not needed since you're using ESLint.

File: packages/itwin/one-click-lca-widget/package.json
Comment (calebmshafer) [inline packages/itwin/one-click-lca-widget/package.json:44]: Why did you need to use cross-fetch, wouldn't just using fetch directly work with this widget?

File: packages/itwin/one-click-lca-widget/src/api/reporting/generated/configuration.ts
Comment (calebmshafer) [inline packages/itwin/one-click-lca-widget/src/api/reporting/generated/configuration.ts:5]: are these added by the auto-generator?

File: packages/itwin/one-click-lca-widget/src/api/reporting/openapi/reporting.json
Comment (calebmshafer) [inline packages/itwin/one-click-lca-widget/src/api/reporting/openapi/reporting.json:10]: This is a duplicate, and different, than the one that was already made over in the [grouping-mapping widget](https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/grouping-mapping-widget/src/api/openapi/reporting.json). Has it been discussed creating a client package that can be shared between the components, and anyone else who wants to use the API?

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: packages/itwin/one-click-lca-widget/src/api/reporting/openapi/reporting.json
Comment (bingjiez) [reply packages/itwin/one-click-lca-widget/src/api/reporting/openapi/reporting.json:10]: We are working on creating a client package first. Will come back to resolve this later

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:2]: I think we need a different package name, this doesn't fit with out current naming schema, itwin/xyz-react

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:5]: add keyword for iTwin.js

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:38]: plz add some tests

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:41]: you shouldnt need to depend on this directly, whats the purpose?

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:44]: why `cross-fetch`?

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:67]: are you guys even using redux?

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:86]: please dont globally disable this across the pkg

Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/package.json:86]: can the jsx-ally rules be fixed?

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:71]: you need a peer dep on core-frontend

File: packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx:21]: do we still not have an actual client that not generated?

File: packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx:43]: string should be localized

File: packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx:73]: I dont think will work once we move to webpack 5

File: packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx:151]: why do all of these have react fragments wrapping divs?

File: packages/itwin/one-click-lca-widget/src/components/Reports.tsx
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/Reports.tsx:10]: if this is the same generated reporting client used in the grouping-mapping widget, then you guys need to increase the priority of getting a standalong client out

**Review by aruniverse — CHANGES_REQUESTED** [review]
Quite a few concerns before we can get this pr merged in and publish this package.
Please leave the comments open for the maintainers to resolve. 
We need to talk regarding these generated clients....
You need to update the rush.json, version-policy.json, codeowners files, lockfile

This PR has a lot to do before we can approve

File: packages/itwin/one-click-lca-widget/src/components/Reports.tsx
Comment (bingjiez) [reply packages/itwin/one-click-lca-widget/src/components/Reports.tsx:10]: We have created insights client for all Insights and Reporting requests

File: packages/itwin/one-click-lca-widget/package.json
Comment (Guillar1) [inline packages/itwin/one-click-lca-widget/package.json:44]: Can you make this a devDependency?

Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/package.json:44]: or remove the dependency altogether? why are you using the css pkg directly?

Comment (Guillar1) [reply packages/itwin/one-click-lca-widget/package.json:44]: https://github.com/iTwin/viewer-components-react/pull/168#discussion_r798040961 
This was talked about before with the G&M widget.

Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/package.json:44]: gah, we need to figure out why this is happening, its def a dep of itwinui-react so should be getting pulled in. lets remove this dep in all our pkgs, run `rush purge, then `rush update --full`, and then manually check the lock file for itwinui-css

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/package.json]: sorry, i know i said add tests, but i think certa here is probably overkill, couldve just gone with some unit tests i think, especially for the little test that you have.

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/package.json:106]: could you update the config here to:

```
  "eslintConfig": {
    "plugins": [
      "@itwin"
    ],
    "extends": "plugin:@itwin/ui",
    "rules": {
      "no-duplicate-imports": "off",
      "@typescript-eslint/consistent-type-imports": "error"
    }
  }
```

File: packages/itwin/one-click-lca-widget/src/components/ExportModal.scss
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/ExportModal.scss:54]: i worry this might break consumers if this is an actual class coming from itwinui

Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/src/components/ExportModal.scss:54]: please update some of the generic class names here so they dont collide

File: packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx:69]: using type `NODEJS.Timer` is unsafe here, this will cause issues in the browser because the node api isnt available there. this will def cause issues in webpack 5 unless you add a polyfill, which is still a bad idea

File: packages/itwin/one-click-lca-widget/README.md
Comment (aruniverse) [inline packages/itwin/one-click-lca-widget/README.md:28]: please remove this section, we should not be exposing this

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/package.json]: @calebmshafer , i dont love the new name either, but thoughts?

File: packages/itwin/one-click-lca-widget/package.json
Comment (Guillar1) [reply packages/itwin/one-click-lca-widget/package.json]: Gave it a try again with the G&M widget. Seems to be working now 🥴.

File: packages/itwin/one-click-lca-widget/package.json
Comment (aruniverse) [reply packages/itwin/one-click-lca-widget/package.json]: how do we feel about one-click-lca-react? still quite a mouthful, so oclca-react?

Comment (arnobmallickbsw) [reply packages/itwin/one-click-lca-widget/package.json]: oclca-react sounds like a good option to me
I doubt people will be looking for the widget without already knowing what one click lca is

**Comment by aruniverse** [comment]
Failing build will hopefully be resolved by https://github.com/iTwin/itwinjs-core/pull/3446

**Comment by bingjiez** [comment]
> Failing build will hopefully be resolved by [iTwin/itwinjs-core#3446](https://github.com/iTwin/itwinjs-core/pull/3446)

When will this be merged? Do we need to wait for it?

**Comment by aruniverse** [comment]
> > Failing build will hopefully be resolved by [iTwin/itwinjs-core#3446](https://github.com/iTwin/itwinjs-core/pull/3446)
> 
> When will this be merged? Do we need to wait for it?

Yes you need it, you need 3.1.1 to be released, to unblock the test failure here

**Comment by aruniverse** [comment]
could you revert the copyright on common/scripts files? and also the .vscode file too?

File: packages/itwin/one-click-lca-widget/src/test/ExportModal.test.tsx
Comment (skirby1996) [inline packages/itwin/one-click-lca-widget/src/test/ExportModal.test.tsx:14]: Should these two expect().toBeDefined lines be different here?

**Review by skirby1996 — CHANGES_REQUESTED** [review]

File: packages/itwin/breakdown-trees/package.json
Comment (MeghaCBorty) [inline packages/itwin/breakdown-trees/package.json:80]: Why do you need to make this change in this package?

Comment (aruniverse) [reply packages/itwin/breakdown-trees/package.json:80]: Security vulnerability. 
I should've caught this last time.

**Review by skirby1996 — APPROVED** [review]

**Pull request merged**
